HerPublicWebsite/Controllers/QuestionnaireController.cs

```diff
@@ -310,6 +312,62 @@ public async Task<IActionResult> ManualAddress_Post(ManualAddressViewModel viewM
         var nextStep = questionFlowService.NextStep(QuestionFlowStep.ManualAddress, questionnaire);
         return RedirectToNextStep(nextStep);
     }
+    
+    [HttpGet("select-local-authority")]
+    public IActionResult SelectLocalAuthority_Get()
+    {
+        var questionnaire = questionnaireService.GetQuestionnaire();
+        var viewModel = new SelectLocalAuthorityViewModel()
+        {
+            BackLink = GetBackUrl(QuestionFlowStep.SelectLocalAuthority, questionnaire)
+        };
+
+        return View("SelectLocalAuthority", viewModel);
+    }
+    
+    [HttpGet("select-local-authority/{custodianCode}")]
+    public IActionResult LocalAuthoritySelected_Get(string custodianCode)
+    {
+        if (!LocalAuthorityData.LocalAuthorityDetailsByCustodianCode.ContainsKey(custodianCode))
+        {
+            // This should only happen if someone messes with the URL
+            logger.LogError("Unrecognised custodian code received: " + custodianCode);
+            return SelectLocalAuthority_Get();
+        }
+        
+        var questionnaire = questionnaireService.UpdateLocalAuthority(custodianCode);

```

Comment:
It is arguably a bit naughty to change state on a GET request, but I tried some other methods and this seems like the best compromise.

Reply:
That is what I had originally, but it made it impossible to correctly fill in the radio button for all case of going back/forwards. It's much simpler if the questionnaire reflects the exact questions in the flow.